tests/e2e/monitoring_test.go

```diff
@@ -1097,3 +1108,456 @@ func (tc *MonitoringTestCtx) ValidateThanosQuerierNotDeployedWithoutMetrics(t *t
+			},
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "ClusterRole",
+			Name:     "data-science-metrics-admin",
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      serviceAccountName,
+				Namespace: namespace,
+			},
+		},
+	}
+
+	tc.EventuallyResourceCreated(WithObjectToCreate(crb))
+	tc.EnsureResourceExists(
+		WithMinimalObject(gvk.ClusterRoleBinding, types.NamespacedName{Name: bindingName}),
+		WithCustomErrorMsg("ClusterRoleBinding should be created for ServiceAccount"),
+	)
+
+	t.Logf("Created ClusterRoleBinding '%s' binding ServiceAccount %s/%s to ClusterRole data-science-metrics-admin",
+		bindingName, namespace, serviceAccountName)
+
+	// Wait for RBAC to propagate
+	time.Sleep(5 * time.Second)
+}
+
+// cleanupTestServiceAccount removes the test ServiceAccount and its ClusterRoleBinding.
+func (tc *MonitoringTestCtx) cleanupTestServiceAccount(t *testing.T, serviceAccountName, namespace string) {
+	t.Helper()
+
+	tc.DeleteResource(
+		WithMinimalObject(gvk.ServiceAccount, types.NamespacedName{Name: serviceAccountName, Namespace: namespace}),
+		WithIgnoreNotFound(true),
+		WithWaitForDeletion(true),
+	)
+
+	bindingName := fmt.Sprintf("test-%s-metrics-binding", serviceAccountName)
+	tc.DeleteResource(
+		WithMinimalObject(gvk.ClusterRoleBinding, types.NamespacedName{Name: bindingName}),
+		WithIgnoreNotFound(true),
+		WithWaitForDeletion(true),
+	)
+}
+
+// getServiceAccountToken retrieves the token for a service account.
+func (tc *MonitoringTestCtx) getServiceAccountToken(t *testing.T, serviceAccountName, namespace string) string {
+	t.Helper()
+
+	tokenRequest := &authenticationv1.TokenRequest{
+		Spec: authenticationv1.TokenRequestSpec{
+			ExpirationSeconds: func() *int64 { i := int64(3600); return &i }(), // 1 hour
+		},
+	}
+
+	ctx := tc.Context()
+	err := tc.Client().SubResource("token").Create(ctx, &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      serviceAccountName,
+			Namespace: namespace,
+		},
+	}, tokenRequest)
+
+	require.NoError(t, err, "Failed to create token for service account")
+	require.NotEmpty(t, tokenRequest.Status.Token, "Token should not be empty")
+
+	return tokenRequest.Status.Token
+}
+
+// testMetricsAccess tests access to the metrics endpoint with the given token.
+func (tc *MonitoringTestCtx) testMetricsAccess(t *testing.T, routeURL, token, namespace string, expectSuccess bool, errorMsg string) {
+	t.Helper()
+
+	// Create HTTP client with custom transport to skip TLS verification for tests
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true}, // #nosec G402 - Required for testing with self-signed certificates
+	}
+	client := &http.Client{Transport: tr}
+	requestURL := fmt.Sprintf("%s/api/v1/query?query=up&namespace=%s", routeURL, namespace)
+	req, err := http.NewRequestWithContext(t.Context(), http.MethodGet, requestURL, nil)
```

Comment:
linked to my other comment: if you change to a POST here, the test should fail because the SA has no CREATE permission (POST is a supported method for many Prometheus API endpoints).